adagrams/game.py

```diff
@@ -1,11 +1,145 @@
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+    }
+
+    def pick_random_letter():
+        # choose a letter randomly from the dict
+        letters = [letter for letter, value in LETTER_POOL.items() if value > 0] # looping through LETTER_POOL to get the value of the letters only if the value is bigger than 0
+        if not letters:
+            return None    #if there are no available letters, return None
+        # use random.randint to pick a letter by index
+        index = random.randint(0, len(letters) - 1)        # generates a random index between 0 and len(letters) - 1
+        return letters[index]
+    
+    draw_letters = []
+
+    for i in range(10):    #select exactly 10 letters from dict
+        letter = pick_random_letter()
+        if letter is None:         # if None is returned indicating no more letters available, break the loop
+            break
+        draw_letters.append(letter)    # adding the randomly picked letters into the initial empty list
+        LETTER_POOL[letter] -= 1       # decrease the count of the drawn letter in the dict
+        if LETTER_POOL[letter] == 0:   # if the count of a letter reaches 0, removes the letter from the dict
+            del LETTER_POOL[letter] # remove the letter if its value reaches 0
+
+    return draw_letters
+
+result = draw_letters()
 
 def uses_available_letters(word, letter_bank):
-    pass
+    # function that counts each letters that occured in the list of letters 
+    # this part of the code can be replaced by import counter
+    def count_letters(letters):
+        counts = {}
+        for letter in letters:
+            if letter in counts:
+                counts[letter] += 1
+            else:
+                counts[letter] = 1
+        return counts
+    
+    # convert word and letter_bank to lowercase to handle case sensitiveness
+    word = word.lower()
+    letter_bank = [letter.lower() for letter in letter_bank]
+    
+    # count the frequencies of the letters in word and letter_bank (word represents the user input words and letter_bank represent the letters that have been drawn)
+    word_counts = count_letters(word)
+    letter_bank_counts = count_letters(letter_bank)
+
+    # check to see if every letters in word is available in letter_bank
+    for letter, count in word_counts.items():
+        if letter_bank_counts.get(letter, 0) < count:
+            return False           # if the letter count in word is more than what we have in letter_bank
+    return True                    # if all the letters are available in letter_bank   
+
 
 def score_word(word):
-    pass
+    # score map, value of each letters
+    letter_scores = {
```

Comment:
This dictionary looks good! Just a heads up that it could also be seen as a global variable that could be used elsewhere!